examples/gno.land/p/demo/forms/create.gno

```diff
@@ -0,0 +1,105 @@
+package forms
+
+import (
+	"std"
+	"time"
+
+	"gno.land/p/demo/json"
+)
+
+func CreateField(label string, fieldType string, required bool) Field {
+	return Field{
+		Label:     label,
+		FieldType: fieldType,
+		Required:  required,
+	}
+}
+
+// CreateForm creates a new form with the given parameters
+func (db *FormDB) CreateForm(title string, description string, openAt string, closeAt string, data string) (string, error) {
+	// Parsing the dates
+	const dateFormat = "2006-01-02T15:04:05Z"
+
+	var parsedOpenTime, parsedCloseTime time.Time
+
+	if openAt != "" {
+		var err error
+		parsedOpenTime, err = time.Parse(dateFormat, openAt)
+		if err != nil {
+			return "", errInvalidDate
+		}
+	}
+
+	if closeAt != "" {
+		var err error
+		parsedCloseTime, err = time.Parse(dateFormat, closeAt)
+		if err != nil {
+			return "", errInvalidDate
+		}
+	}
+
+	// Parsing the json submission
+	node, err := json.Unmarshal([]byte(data))
+	if err != nil {
+		return "", errInvalidJson
+	}
+
+	fieldsCount := node.Size()
+	fields := make([]Field, fieldsCount)
+
+	// Parsing the json submission to create the gno data structures
+	for i := 0; i < fieldsCount; i++ {
+		field, err := node.GetIndex(i)
+		if err != nil {
+			return "", errInvalidJson
+		}
+
+		labelNode, err := field.GetKey("label")
+		if err != nil {
+			return "", errInvalidJson
+		}
+		fieldTypeNode, err := field.GetKey("fieldType")
+		if err != nil {
+			return "", errInvalidJson
+		}
+		requiredNode, err := field.GetKey("required")
+		if err != nil {
+			return "", errInvalidJson
+		}
+
+		label, err := labelNode.GetString()
+		if err != nil {
+			return "", errInvalidJson
+		}
+		fieldType, err := fieldTypeNode.GetString()
+		if err != nil {
+			return "", errInvalidJson
+		}
+		required, err := requiredNode.GetBool()
+		if err != nil {
+			return "", errInvalidJson
+		}
+
+		fields[i] = CreateField(label, fieldType, required)
+	}
```

Comment:
In this case, it can be handled thought method chaing using `MustKey` and `MustT` (where `T` is an aribitary type). Although it causes a panic instead of error, this won't occur if the JSON is in the correct form.

```go
for i := 0; i < fieldsCount; i++ {
    field := node.MustIndex(i)
    
    fields[i] = CreateField(
        field.MustKey("label").MustString(),
        field.MustKey("fieldType").MustString(),
        field.MustKey("required").MustBool(),
    )
}
```